lib/puppetserver/ca/generate_action.rb

```diff
@@ -0,0 +1,249 @@
+require 'optparse'
+require 'openssl'
+require 'puppetserver/utils/file_utils'
+require 'puppetserver/settings/ttl_setting'
+
+module Puppetserver
+  module Ca
+    class GenerateAction
+      CA_EXTENSIONS = [
+        ["basicConstraints", "CA:TRUE", true],
+        ["keyUsage", "keyCertSign, cRLSign", true],
+        ["subjectKeyIdentifier", "hash", false],
+        ["authorityKeyIdentifier", "keyid:always", false]
+      ]
+      DEFAULT_SIGNING_DIGEST = OpenSSL::Digest::SHA256.new
+
+      SUMMARY = "Generate a root and intermediate signing CA for Puppet Server"
+      BANNER = <<-BANNER
+Usage:
+  puppetserver ca generate [--help|--version]
+  puppetserver ca generate [--config PATH]
+
+Description:
+Generate a root and intermediate signing CA for Puppet Server
+and store generated CA keys, certs, and crls on disk.
+
+To determine the target location, the default puppet.conf
+is consulted for custom values. If using a custom puppet.conf
+provide it with the --config flag
+Options:
+BANNER
+
+      def initialize(logger)
+        @logger = logger
+      end
+
+      def run(input)
+        # Validate config_path provided
+        config_path = input['config']
+        errors = Puppetserver::Utils::FileUtilities.validate_file_paths([config_path])
```

Comment:
Sorry, I wrote a novella in the definition of validate_file_paths. tl;dr, I think it would be better to have validate_file_paths no accept nil values at all and wrap this validation in an `if config_path` block.

Reply:
done!  Thanks